File: Algorithms/Sorting/Quicksort_1-Partition/main.py
### Quicksort 1 - Partition - Solution

def quickSort(arr):
    pivotNum = arr[0]
    for i in range(1, len(arr)):
        if pivotNum > arr[i]:
            for j in range(i, 0, -1):
                temp = arr[j]
                arr[j] = arr[j-1]
                arr[j-1] = temp
    print(*arr)

n = int(input())
arr = list(map(int, input().split()[:n]))
quickSort(arr)

File: Algorithms/Sorting/Quicksort_1-Partition/main.cpp
// Quicksort 1 - Partition - Solution

#include <iostream>

void quickSort(int n, int arr[]) {
    int pivotNum{arr[0]}, temp;
    for (int i{1}; i < n; i++) {
        if (pivotNum > arr[i]) {
            for (int j{i}; j > 0; j--) {
                temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
            }
        }
    }
    for (int k{}; k < n; k++)
        std::cout << arr[k] << " ";
}

int main() {
    int n;
    std::cin >> n;
    int arr[n];
    for (int i{}; i < n; i++) 
        std::cin >> arr[i];
    quickSort(n, arr);

    return 0;
}